app/d3plus.js

```diff
@@ -95,6 +98,75 @@ function findColor(d) {
   return Object.keys(d).some(v => badMeasures.includes(v)) ? bad : good;
 }
 
+function findColorTooltip(key, d) {
+  let detectedColors = [];
+  if (this && this._filteredData) {
+    detectedColors = Array.from(new Set(this._filteredData.map(findColor)));
+  }
+  if (detectedColors.length !== 1) {
+    if (colors[key]) {
+      const color = key === "NAPCS" ? colors[key] : colors[key][`${d[`ID ${key}`]}`] || colors[key][`${d[key]}`] || colors[key][`${d[`${key} ID`]}`];
+      if (color) return color;
+    }
+    else if (listGeo.includes(key)) {
+      const color = colors.Geo;
+      if (color) return color;
+    }
+    return colors.colorGrey;
+  }
+  return Object.keys(d).some(v => badMeasures.includes(v)) ? bad : good;
+}
+
+const getTooltipTitle = (d3plusConfig, d) => {
+  const len = Array.isArray(d3plusConfig._groupByRaw) ? d3plusConfig._groupByRaw.length : 1;
+
+  const parentName = Array.isArray(d3plusConfig._groupByRaw) ? d3plusConfig._groupByRaw[0] : d3plusConfig._groupByRaw;
+
+  let parent = Object.entries(d).find(h => h[0] === parentName)[1] || [undefined];
+  let parentId = parentName;
+  if (parentId.includes("ID")) {
+    parentId = parentId.replace("ID", "").trim();
+    parent = Object.entries(d).find(h => h[0] === parentId)//[1] || [undefined];
+    parent = parent && parent[1] ? parent[1] : [undefined]
+  }
+  const itemName = Array.isArray(d3plusConfig._groupByRaw) ? d3plusConfig._groupByRaw[len - 1] : d3plusConfig._groupByRaw;
+
+  let item = Object.entries(d).find(h => h[0] === itemName)[1] || [undefined];
+  let itemId = itemName;
+  if (itemId.includes("ID")) {
+    itemId = itemId.replace("ID", "").trim();
+    item = Object.entries(d).find(h => h[0] === itemId)//[1] || [undefined];
+    item = item && item[1] ? item[1] : [undefined]
+  }
+
+  return {item, itemId, parent, parentId};
+};
+
+export const tooltipTitle = (bgColor, imgUrl, title, subtitle=false) => {
+  let tooltip = "<div class='d3plus-tooltip-title-wrapper'>";
+
+  if (imgUrl) {
+    tooltip += `<div class="icon" style="background-color: ${bgColor}"><img src="${imgUrl}" /></div>`;
+  }
+
+  tooltip += `<div class=${imgUrl ? "title" : "title-without-icon"}><span>${title}</span></div>`;
+
+  if(subtitle){
+    tooltip += `<div class="subtitle"><span>${subtitle}</span></div>`;
+  }
+  tooltip += "</div>";
+  return tooltip;
+};
+
+export const findIconV2 = (key, d) => {
```

Comment:
I know this was copied from another project... but can we just call this `findIcon`? 😆